File: TestProject1/PageObjects/MainForm.cs
using OpenQA.Selenium.Appium.Windows;

namespace WinAppDriverTests
{
    public class MainForm
    {
        private readonly WindowsDriver<WindowsElement> _driver;

        public MainForm(WindowsDriver<WindowsElement> driver)
        {
            _driver = driver;
        }

        public WindowsElement HelloWorldButton => _driver.FindElementByName("Click me");

        public MessageBoxForm ClickHelloWorldButton()
        {
            HelloWorldButton.Click();
            return new MessageBoxForm(_driver);
        }
    }
}

File: CucumberTest/src/test/java/com/example/pageobject/MainForm.java
package com.example.pageobject;

import org.openqa.selenium.WebElement;

import com.example.steps.ApplicationBootSteps;

import io.appium.java_client.windows.WindowsDriver;

public class MainForm {
    private WindowsDriver<WebElement> _driver;

    public MainForm() {
        _driver = ApplicationBootSteps.getDriver();
    }

    public void clickClickMeButton() {
        WebElement clickMeButton = _driver.findElementByName("Click me");
        clickMeButton.click();
    }

    public void clickCallApiButton() {
        WebElement callApiButton = _driver.findElementByName("Call Api");
        callApiButton.click();
    }
}
